tests/backup/backup_resiliency_test.go

```diff
@@ -1298,3 +1281,263 @@ var _ = Describe("{RebootNodesWhenBackupsAreInProgress}", func() {
+			originalReplicaCount = *backupDeployment.Spec.Replicas
+			log.Infof("Replica count for px-backup pod before taking backup is %v", originalReplicaCount)
+			log.FailOnError(err, fmt.Sprintf("Getting px-backup deployment replica in backup namespace %s", pxBackupNS))
+			dash.VerifyFatal(originalReplicaCount > scaledDownReplica, true, "Verifying px_backup deployment replica before taking backup")
+		})
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			ctx, err = backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					sem <- struct{}{}
+					backupName := fmt.Sprintf("%s-%s-%d-%v", BackupNamePrefix, namespace, i, time.Now().Unix())
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string, namespace string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						_, err = CreateBackupWithoutCheck(backupName, SourceClusterName, bkpLocationName, backupLocationUID,
+							[]string{namespace}, labelSelectors, orgID, srcClusterUid, "", "", "", "", ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup %s of application- %s", backupName, namespace))
+					}(backupName, namespace)
+				}
+			}
+			wg.Wait()
+			log.InfoD("The list of backups taken are: %v", backupNames)
+		})
+		Step("Scaling px-backup deployment replica count to 0 and back to original replica while backup is in progress", func() {
+			log.InfoD("Scaling px-backup deployment replica count to 0 while backup is in progress")
+			*backupDeployment.Spec.Replicas = scaledDownReplica
+			backupDeployment, err = apps.Instance().UpdateDeployment(backupDeployment)
+			log.FailOnError(err, fmt.Sprintf("Scaling down px-backup deployment replica to 0"))
+			log.Infof("px-backup replica count after scaling to 0 is %v", *backupDeployment.Spec.Replicas)
+			dash.VerifyFatal(*backupDeployment.Spec.Replicas == scaledDownReplica, true, "Verify px-backup deployment replica after scaling down")
+			log.InfoD("Sleeping for 10 minute while backup is in progress and px-backup deployment's replica count is 0")
+			time.Sleep(10 * time.Minute)
+			log.InfoD("Scaling px-backup deployment to original replica while backup is in progress")
+			backupDeployment, err = apps.Instance().GetDeployment(pxBackupDeployment, pxBackupNS)
+			*backupDeployment.Spec.Replicas = originalReplicaCount
+			backupDeployment, err = apps.Instance().UpdateDeployment(backupDeployment)
+			log.FailOnError(err, fmt.Sprintf("Scaling px_backup deployment back to original replica"))
+			log.Infof("px-backup replica after scaling back to original replica is %v", *backupDeployment.Spec.Replicas)
+			dash.VerifyFatal(*backupDeployment.Spec.Replicas == originalReplicaCount, true, "Verify px-backup deployment replica after scaling back to original")
```

Comment:
How about dash.VerifyFatal(*backupDeployment.Spec.Replicas,  originalReplicaCount, "Verify px-backup deployment replica after scaling back to original")?